sda/protocol.go

```diff
@@ -37,59 +38,62 @@ type ProtocolInstance interface {
 	Shutdown() error
 }
 
-// NewProtocol is the function-signature needed to instantiate a new protocol
-type NewProtocol func(*TreeNodeInstance) (ProtocolInstance, error)
+var protocols = NewProtocolStorage()
 
-// ProtocolNameToID returns the ProtocolID corresponding to the given name
-func ProtocolNameToID(name string) ProtocolID {
-	url := network.NamespaceURL + "protocolname/" + name
-	return ProtocolID(uuid.NewV3(uuid.NamespaceURL, url))
+// ProtocolStorage holds all protocols either globally or per-Conode.
+type ProtocolStorage struct {
+	// Instantiators maps the name of the protocols to the `NewProtocol`-
+	// methods.
+	Instantiators map[string]NewProtocol
```

Comment:
Is there a reason why it's public ? Otherwise I would prefer this being unexported, otherwise it's possible to mess up things pretty bad... :D